### PR TITLE
improve apt tasks

### PR DESCRIPTION
**Pull request by kwypchlo**
- removed standalone "apt-get update" tasks in favour of updating apt-get when running a task that installs a package
- replaced "loop" with a "name" list
- increased "cache_valid_time" from 15 minutes to 30 minutes to increase the chance that we will only have to update once since most jobs should finish in 30 minutes (we can increase it to 60 minutes too fwiw)
- replaced "state: present" with "state: latest" to make sure packages are up to date
- unfortunately we can't pin specific versions of packages since linux distro version will have a little bit different package versions (for example different versions of ubuntu have different available tmux versions)

### CODE REVIEW

**Comment by MSevey** [comment]
> unfortunately we can't pin specific versions of packages since linux distro version will have a little bit different package versions (for example different versions of ubuntu have different available tmux versions)

We currently have a lock on the linux distros for the portals though. Debian 10 and debian 11. So shouldn't we be able to set the apt package versions?

**Comment by MSevey** [comment]
if we are cleaning this up, we should clean it up all the way, which would include refactoring it into a helper tasks so that the options of `apt` are only defined once and is consistent across plays.

**Comment by kwypchlo** [comment]
> > unfortunately we can't pin specific versions of packages since linux distro version will have a little bit different package versions (for example different versions of ubuntu have different available tmux versions)
> 
> We currently have a lock on the linux distros for the portals though. Debian 10 and debian 11. So shouldn't we be able to set the apt package versions?

we also support ubuntu but unfortunately even debian 10 and 11 have different packages available on apt

**Comment by MSevey** [comment]
> > > unfortunately we can't pin specific versions of packages since linux distro version will have a little bit different package versions (for example different versions of ubuntu have different available tmux versions)
> > 
> > 
> > We currently have a lock on the linux distros for the portals though. Debian 10 and debian 11. So shouldn't we be able to set the apt package versions?
> 
> we also support ubuntu but unfortunately even debian 10 and 11 have different packages available on apt

sad.

well then we should still refactor all these apt calls into a task so that they all are handled the same way in terms of cache updates. That will make updates in the future easier.

File: playbooks/deploy-machines-setup.yml [hunk 1]
```diff
@@ -25,22 +25,16 @@
...
       include_tasks: tasks/check-server-os-compatibility.yml
 
-    # Configure server
-    - name: Run "apt-get update"
-      apt:
-        update_cache: yes
-        cache_valid_time: 900
-      become: True
-
     - name: Install packages
```
Comment (MSevey) [inline playbooks/deploy-machines-setup.yml:28]: We should refactor this out in a helper task that accepts a list of packages. 

then we ensure the `update_cache cache_valid_time state` values are always the same for every call

**Review by MSevey — CHANGES_REQUESTED** [review]

File: playbooks/deploy-machines-setup.yml (hunk 1 above)
Comment (kwypchlo) [reply playbooks/deploy-machines-setup.yml:28]: @MSevey maybe you would like to reconsider your change request, this is pretty good improvement as is and making it a helper task could be resolved in potential follow up

Comment (MSevey) [reply playbooks/deploy-machines-setup.yml:28]: yea fine as a follow up

**Review by MSevey — APPROVED** [review]

File: playbooks/deploy-machines-setup.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kwypchlo) [reply playbooks/deploy-machines-setup.yml:28]: follow up in https://github.com/SkynetLabs/ansible-playbooks/issues/344

**Review by firyx — APPROVED** [review]

**Pull request merged**